storage/innobase/include/arch0arch.h

```diff
@@ -0,0 +1,1984 @@
+
+  /** Checks if memory range is all zeros.
+  @param[in]  start            The pointer to first byte of a buffer
+  @param[in]  number_of_bytes  The number of bytes in the buffer
+  @return true if and only if `number_of_bytes` bytes pointed by
+  `start` are all zeros. */
+  static bool is_zeros(const void *start, size_t number_of_bytes);
+
+  /** Get block type from the block header.
+  @param[in]     block   block from where to get the type
+  @return block type */
+  static Arch_Blk_Type get_type(byte *block);
+
+  /** Get block data length from the block header.
+  @param[in]     block   block from where to get the data length
+  @return block data length */
+  static uint get_data_len(byte *block);
+
+  /** Get the stop lsn stored in the block header.
+  @param[in]     block   block from where to fetch the stop lsn
+  @return stop lsn */
+  static lsn_t get_stop_lsn(byte *block);
+
+  /** Get the block number from the block header.
+  @param[in]     block   block from where to fetch the block number
+  @return block number */
+  static uint64_t get_block_number(byte *block);
+
+  /** Get the reset lsn stored in the block header.
+  @param[in]     block   block from where to fetch the reset lsn
+  @return reset lsn */
+  static lsn_t get_reset_lsn(byte *block);
+
+  /** Get the checksum stored in the block header.
+  @param[in]     block   block from where to fetch the checksum
+  @return checksum */
+  static uint32_t get_checksum(byte *block);
+
+  /** Fetch the offset for a block in the archive file.
+  @param[in]    block_num       block number
+  @param[in]    type            type of block
+  @return file offset of the block */
+  static uint64_t get_file_offset(uint64_t block_num, Arch_Blk_Type type);
+
+ private:
+  /* @note member function copy_data needs to be updated whenever a new data
+  member is added to this class. */
+
+  /** Block data buffer */
+  byte *m_data;
+
+  /** Block data length in bytes */
+  uint m_data_len{};
+
+  /** Total block size in bytes */
+  uint m_size;
+
+  /** State of the block. */
+  Arch_Blk_State m_state{ARCH_BLOCK_INIT};
+
+  /** Unique block number */
+  uint64_t m_number{};
+
+  /** Type of block. */
+  Arch_Blk_Type m_type;
+
+  /** Checkpoint lsn at the time the last page ID was added to the
+  block. */
+  lsn_t m_stop_lsn{LSN_MAX};
+
+  /** Oldest LSN of all the page IDs added to the block since the last
+   checkpoint */
+  lsn_t m_oldest_lsn{LSN_MAX};
+
+  /** Start LSN or the last reset LSN of the group */
+  lsn_t m_reset_lsn{LSN_MAX};
+};
+
+/** Archiver file context.
+Represents a set of fixed size files within a group */
+class Arch_File_Ctx
+{
+ public:
+  class Recovery;
+
+  /** Constructor: Initialize members */
+  Arch_File_Ctx() { m_file.m_file = OS_FILE_CLOSED; }
+
+  /** Destructor: Close open file and free resources */
+  ~Arch_File_Ctx()
+  {
+    close();
+    if (m_name_buf)
+      ut_free(m_name_buf);
+  }
```

Comment:
Unreachable class. The `if` is redundant, and we could use a default constructor if we used an initializer expression for `m_file`.